File: pgmeascommons/src/main/java/it/csi/pgmeas/commons/model/Fase.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2025 Regione Piemonte
*
* SPDX-License-Identifier: EUPL-1.2 
*/
package it.csi.pgmeas.commons.model;

import java.io.Serializable;

import it.csi.pgmeas.commons.model.base.DecodificaBaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name="pgmeas_d_fase")
public class Fase extends DecodificaBaseEntity implements Serializable {

	private static final long serialVersionUID = -6670936011587388137L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "fase_id")
	private Integer faseId;
	
	@Column(name = "fase_cod")
	private String faseCod;
	
	@Column(name = "fase_desc")
	private String faseDesc;
	
}

File: pgmeasapp/projects/pgmeas-library/model/src/fase.ts
/*
* SPDX-FileCopyrightText: (C) Copyright 2025 Regione Piemonte
*
* SPDX-License-Identifier: EUPL-1.2 
*/

import { Tipologica } from "./tipologica";

export interface Fase extends Tipologica {
  faseId: number;
  faseCod: string;
  faseDesc: string;
}
